### DISCUSSION TITLE
Applying default settings in React

### DISCUSSION THREAD

**Discussion by scarf005** [tabler/tabler-icons#307] [Q&A] [answered]
hello, awesome library. One thing I'm wondering would there be any way to apply default settings for all icons?
for example, i'd wish every icons to have `stroke-width` of 1.25 so I don't have to do `<IconPlus stroke={1.25} />` for every imported icons.

**Comment by ghost** [comment]
If you mean a config, I don't think so, but you can create a file where you'll be exporting your default props and share them across your app, for example:
```js
/** styles/index.ts */

export const iconProps = {
    size: 18,
    stroke: 1.25
}
```

```js
/** Component.tsx */

import { IconPlus } from "@tabler/icons"
import { iconProps } from "styles"

export default Component = () => {
    return <IconPlus {...iconProps} />
}

```

**Comment by ShahriarKh** [comment]
[react-loading-skeleton](https://github.com/dvtng/react-loading-skeleton#theming) has something called `<SkeletonTheme />` to apply settings to all `<Skeleton />` children. Implement something similar sounds good:
```jsx
<IconTheme size={18} stroke={1.25}>
   <IconPlus />
   <IconMinus />
</IconTheme>
```

**Comment by sarendipitee** [comment, marked as answer]
For anyone coming across this from google, you can override the default with a global css selector:
```
svg.tabler-icon {
    stroke-width: 1;
    width: 16px;
    height: 16px;
 }
 ```
